app/src/main/java/com/nextcloud/talk/dagger/modules/DatabaseModule.java

```diff
@@ -44,7 +46,13 @@ public class DatabaseModule {
 
     @Provides
     @Singleton
-    public SqlCipherDatabaseSource provideSqlCipherDatabaseSource(@NonNull final Context context) {
+    public SqlCipherDatabaseSource provideSqlCipherDatabaseSource(
+        @NonNull final Context context,
+        final AppPreferences appPreferences) {
+        int version = DB_VERSION;
+        if (appPreferences.getIsDbRoomMigrated()) {
+            version++;
+        }
```

Comment:
this is needed due to the upgrade vs. new install scenario you faced. So initially it needs to be version 7 based on requery, then Room updates it to version 8, then on any future start of the app requery when loading the database needs to also request version 8. But if requery does v8 than the db will be at v8 and Room's upgrade won't run because the Db is at v8 but then Room will crash because the structure hasn't been migrated.

And yes, this is a one-way street with very strict road: 
**no db changes before having removed all uses of requery to access the database**

After that we should keep the framework and db initialization in for a bit, since requery did the upgrades to v7, unfortunately we would probably need to spent some deep digging to find out the differences between the version. If we have them, then it is easier since then Room could do all the migrations.

So my idea was, to wait until we have everything on Room, then wait for older versions to have very little to no usage in the gplay reports, then throw out requery completely.